Update core dependencies

jdillon opened the pull request:
sisu/guice/guava & related.

jdillon committed:
Use ${jackson.version} used in more than 2 places.

jdillon committed:
Use javax.annotation.Nullable, tidy.

jdillon committed:
Add goodies DM

jdillon committed:
Use latest sisu-bundle-launcher 1.5-SNAPSHOT

jdillon committed:
Tidy, drop uneeded exclusions.  Add jsr305, notes

jdillon committed:
Use DM for goodies

jdillon committed:
Pick up jsr305

jdillon committed:
Latest plexus-utils 3.0.8

jdillon committed:
Add DM for more sisu/guice bits. Tidy and docs

jdillon commented:
will link to grid build on success.

jdillon commented:
https://builds.sonatype.org/job/nexus-oss-its-feature/470/ lnx=SUCCESS win=FAILED 

1 failure in DefaultFSLocalRepositoryStoragePerformance2IT.testRetieveItemWithLastAccessUpdate, which I think this should sort out:

https://github.com/sonatype/nexus/pull/622

IIUC this is unrelated to the dependency changes.

peterlynch commented:
+1

peterlynch commented on nexus-runtime-platform/pom.xml:
```diff
@@ -23,115 +23,114 @@
   </parent>
 
+  <!--
+  FIXME: This does not actually provide deps on the Nexus runtime, but only on the sisu/guice/plexus and basic support libraries.
```
nexus runtime means ' what common bits does nexus runtime need at runtime'

jdillon replied:
Fair enough, though I don't think that really warrants a separate top-level pom just for that... but perhaps its needed by something in pro?

cstamas replied:
Yes. Its used to align sisu and other deps as we had problems with that
before

Thanks,
~t~ mobile
On Oct 26, 2012 8:26 PM, "Jason Dillon" notifications@github.com wrote:

> In nexus-runtime-platform/pom.xml:
> 
> > @@ -23,115 +23,114 @@
> >      <relativePath />
> >    </parent>
> > -  <!--
> > -  FIXME: This does not actually provide deps on the Nexus runtime, but only on the sisu/guice/plexus and basic support libraries.
> 
> Fair enough, though I don't think that really warrants a separate
> top-level pom just for that... but perhaps its needed by something in pro?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/sonatype/nexus/pull/619/files#r1956129.

cstamas replied:
Forgot to add: align them between oss and pro. Just as JD said

Thanks,
~t~ mobile
On Oct 26, 2012 9:39 PM, "Tamás Cservenák" tamas@cservenak.net wrote:

> Yes. Its used to align sisu and other deps as we had problems with that
> before
> 
> Thanks,
> ~t~ mobile
> On Oct 26, 2012 8:26 PM, "Jason Dillon" notifications@github.com wrote:
> 
> > In nexus-runtime-platform/pom.xml:
> > 
> > > @@ -23,115 +23,114 @@
> > >      <relativePath />
> > >    </parent>
> > > -  <!--
> > > -  FIXME: This does not actually provide deps on the Nexus runtime, but only on the sisu/guice/plexus and basic support libraries.
> > 
> > Fair enough, though I don't think that really warrants a separate
> > top-level pom just for that... but perhaps its needed by something in pro?
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/sonatype/nexus/pull/619/files#r1956129.

jdillon committed:
Update plexus-interpolation to 1.15, drop uneeded excludes, drop uneeded deps which are picked up transitively

jdillon committed:
Update plexus-archiver (only used by test framework).
Update plexus-cipher (which is really now sisu, and dropped uneeded exclusions).
Update plexus-cli (only used by migration tool stuff).

jdillon committed:
base-configuration does not depend on plexus, drop excludes.
move plexus-cli and plexus-archiver into module groups where used.

jdillon committed:
notes

jdillon committed:
Merge branch 'master' into update-core-dependencies

jdillon commented:
Another win-only failure:

https://builds.sonatype.org/job/nexus-oss-its-feature/471/

org.sonatype.nexus.proxy.walker.FixedRateWalkerThrottleControllerTest.testDoesItHelpAtAll
